pennylane/ops/functions/equal.py

```diff
@@ -634,7 +636,7 @@ def _equal_hamiltonian(op1: Hamiltonian, op2: Observable, **kwargs):
         return f"{op2} is not of type Observable"
 
     if not op1.compare(op2):
-        return f"'{op1}' and '{op2}' are not same"
+        return f"'{op1}' and '{op2}' are not the same"
 
     return True
```

Comment:
Maybe the same `return op1.compare(op2) or f"'{op1}' and '{op2}' are not the same for an unspecified reason"` here too for readabilities sake?

Reply:
It's not exactly "unspecified reason", the reason here is that the two ops are not mathematically equivalent (which is what `compare()` checks. How about I update it to say that?

Reply:
I guess I included "for an unspecified reason", because `compare` doesn't really tell you why they are not equal.

Reply:
Makes sense, I can do that